toolbox/strings.py

```diff
@@ -102,6 +110,76 @@ def is_invite(string: str, *, fullmatch: bool = True) -> bool:
     return False
 
 
+class MarkdownFormat(IntFlag):
+    NONE = 0
+    STRIKETHROUGH = 1
+    ITALIC_UNDERSCORE = 2
+    ITALIC_ASTERISK = 4
+    BOLD = 8
+    UNDERLINE = 16
+    CODE_BLOCK = 32
+    MULTI_CODE_BLOCK = 64
+    QUOTE = 128
+    MULTI_QUOTE = 256
+    SPOILER = 512
+    ALL = (
+        STRIKETHROUGH
+        | ITALIC_UNDERSCORE
+        | ITALIC_ASTERISK
+        | BOLD
+        | UNDERLINE
+        | CODE_BLOCK
+        | MULTI_CODE_BLOCK
+        | QUOTE
+        | MULTI_QUOTE
+        | SPOILER
+    )
+
+
+format_dict = {
+    MarkdownFormat.MULTI_QUOTE: [r"\s*\>>> ([\s\S]+?)", ">>> {0}"],
+    MarkdownFormat.QUOTE: [r"\s*\> ([\s\S]+?)", "> {0}"],
+    MarkdownFormat.MULTI_CODE_BLOCK: [r"`{3}([\S\s]+?)`{3}", "```{0}```"],
+    MarkdownFormat.CODE_BLOCK: [r"`([^`]+?)`", "`{0}`"],
+    MarkdownFormat.BOLD: [r"\*{2}([\s\S]+?)\*{2}", "\*\*{0}\*\*"],
+    MarkdownFormat.UNDERLINE: [r"__([\s\S]+?)__", "__{0}__"],
+    MarkdownFormat.STRIKETHROUGH: [r"~~([\S\s]+?)~~", "~~{0}~~"],
+    MarkdownFormat.ITALIC_UNDERSCORE: [r"_([^_]+?)_", "_{0}_"],
+    MarkdownFormat.ITALIC_ASTERISK: [r"\*([^*]+?)\*", "\*{0}\*"],
+    MarkdownFormat.SPOILER: [r"\|{2}([\s\S]+?)\|{2}", "\|\|{0}\|\|"],
+}
+
+
+def remove_markdown(content: str, formats: MarkdownFormat = MarkdownFormat.ALL) -> str:
+    """Removes the markdown formatting from Discord messages.
+
+    Parameters
+    ----------
+    content : str
+        The `str` object, which needs their content cleaned from Discord's markdown formatting.
+    formats : MarkdownFormat
+        The `IntFlag` of the formatting that needs to be removed. Default is ALL.
+
+    Returns
+    -------
+    str
+        The cleaned string without markdown formatting.
+    """
+    match_list = []
+    for format in format_dict:
+        if formats & format:
+            search = re.compile(format_dict[format][0])
```

Comment:
It should be more effective to compile the regex in the format_dict

Reply:
Also done, saves on variable in the for loop now as well. Tests still passing